src/NServiceBus.Core/Sagas/SagaPersistenceBehavior.cs

```diff
@@ -13,16 +13,9 @@
 
     class SagaPersistenceBehavior : HandlingStageBehavior
     {
-        ISagaPersister sagaPersister;
-        ICancelDeferredMessages timeoutCancellation;
-        SagaMetadataCollection sagaMetadataCollection;
-
-        public SagaPersistenceBehavior(ISagaPersister persister, ICancelDeferredMessages timeoutCancellation, SagaMetadataCollection sagaMetadataCollection)
```

Comment:
The alternative is using `builder.BuildAll<T>().FirstOrDefault()`. Chose your poison